app/backfill-synonym-slugs.ts

```diff
@@ -0,0 +1,55 @@
+import { tables } from '@architect/functions'
+import type { DynamoDBDocument } from '@aws-sdk/lib-dynamodb'
+import { paginateScan } from '@aws-sdk/lib-dynamodb'
+import { slug } from 'github-slugger'
+
+import type { Synonym } from '~/routes/synonyms/synonyms.lib'
+
+function* chunks<T>(arr: T[], n: number): Generator<T[], void> {
+  for (let i = 0; i < arr.length; i += n) {
+    yield arr.slice(i, i + n)
+  }
+}
+
+export async function backfillSynonyms() {
+  console.log('Starting SYNONYM SLUG backfill...')
+  const db = await tables()
+  const client = db._doc as unknown as DynamoDBDocument
+  const TableName = db.name('synonyms')
```

Comment:
Architect's service discovery will not work in this context. You'll need to look up the name of the table using Amazon SSM.